usr/share/rear/finalize/Linux-ppc64le/620_install_grub2.sh

```diff
@@ -48,9 +48,17 @@ if [[ -r "$LAYOUT_FILE" ]]; then
         if [ -n "$part" ]; then
             LogPrint "Boot partition found: $part"
             dd if=/dev/zero of=$part
+
+            # Do not update nvram when system is running in PowerNV mode (BareMetal).
+            # grub2-install will failed if not run with the --no-nvram option on a PowerNV system.
+            # see https://github.com/rear/rear/pull/1742
+            if [[ $(awk '/platform/ {print $NF}' < /proc/cpuinfo) == PowerNV ]] ; then
```

Comment:
I cannot reproduce this condition on my x86_64 system (there is no 'platform' in /proc/cpuinfo)
but what I can reproduce there is e.g.:
<pre>
# awk '/vendor_id/ {print $NF}' < /proc/cpuinfo
GenuineIntel
GenuineIntel
GenuineIntel
GenuineIntel
GenuineIntel
GenuineIntel
GenuineIntel
GenuineIntel
</pre>
I get 8 lines because I have a 2 CPUs with 4 cores each
but when there is more than one line found
the condition does no longer result true:
<pre>
# if [[ $( awk '/vendor_id/ {print $NF}' < /proc/cpuinfo ) == GenuineIntel ]] ; then echo Y ; else echo N ; fi
N
</pre>
Bottom line:
The current condition is not fail-safe against multiple 'platform' lines.

Usually I use an artificial bash array to get a field at a know position
out of the first line of possibly several output lines like:
<pre>
# vendor_id_lines=( $( grep '^vendor_id' /proc/cpuinfo ) )

# for element in "${vendor_id_lines[@]}" ; do echo "'$element'" ; done
'vendor_id'
':'
'GenuineIntel'
'vendor_id'
':'
'GenuineIntel'
'vendor_id'
':'
'GenuineIntel'
'vendor_id'
':'
'GenuineIntel'
'vendor_id'
':'
'GenuineIntel'
'vendor_id'
':'
'GenuineIntel'
'vendor_id'
':'
'GenuineIntel'
'vendor_id'
':'
'GenuineIntel'

# if test "${vendor_id_lines[2]}" = "GenuineIntel" ; then echo Y ; fi
Y
</pre>
But this only works when the wanted value
is at a known fixed position/index in the array. 
E.g. it cannot be used to test for one of the many CPU flags.

Reply:
@jsmeix ,

on Power, platform is printed only **one time** at the end:
```
processor	: 0
cpu		: POWER8 (raw), altivec supported
clock		: 2061.000000MHz
revision	: 2.0 (pvr 004d 0200)

[...]

processor	: 152
cpu		: POWER8 (raw), altivec supported
clock		: 2061.000000MHz
revision	: 2.0 (pvr 004d 0200)

timebase	: 512000000
platform	: PowerNV
model		: 8001-22C
machine		: PowerNV 8001-22C
firmware	: OPAL v3
```

We can also protect it by checking only the first line returned:
```
if [[ $(awk '/platform/ {print $NF}' < /proc/cpuinfo | head -n1 ) == PowerNV ]] 
```

Reply:
@schabrolles 
feel free to implement it as you like.
When it can appear at most one time it is also o.k. as is.